### PR TITLE
Yet Another Switch to CommonJS

### PR DESCRIPTION
**Pull request by davidlehn**
This is another attempt at a CommonJS switch.  It's similar to #357 but attempts to change the code structure less and adds testings updates.

- [x] Minimal churn from the old style.  Mostly a removal of boilerplate and adapting to CommonJS style.  This was to avoid changing everything at once.  Things can be further refactored once the first major switch happens.  Have `requires()` build up the forge object may be a questionable style but lets address that later.
- [x] Top-level forge object no longer can be called with new options.  There is now a single `forge.options` object.  For now only including the renamed `disableNativeCode` as `usePureJavaScript`.
- [x] Browser bundles built with webpack.  Fairly easy to adapt to just build what you want.  It *could* build bundles for every file, but that makes little sense.  Better to let people build what they need which now works (I hope).
- [x] Browserify testing.  It worked at some point and has some support in the package.json files but hasn't been tested recently.
- [x] Testing: Node.js
- [x] Testing: In a browser manually via a webpack build of all tests with forge
- [x] Testing: PhantomJS, Chrome, Firefox, and Safari via Karma (webpack version)
- [x] Testing: PhantomJS, Chrome, Firefox, and Safari via Karma (browserify version)
- [x] Testing: Travis-CI passing Node.js and PhantomJS tests
- [x] Docs updated
- [x] Some style and misc cleanups
- [x] Bower support. I'm totally against adding the built files to the main repo.  We may need to have a secondary repo just to store built files.
  - Plan is a new forge-dist repo for built bower/bundle support. 0.6.x code will be imported so old checkouts should continue to work the same.  And new 0.7.x checkouts will be totally different but should work for many uses.  Custom cases will require a new workflow of building your own bundle.
- [x] Update files in `./tests/`
- [x] Add docs on various tools and methods to create custom bundles
- [x] Move testing files from `./nodejs` to root dir
- [x] prime.worker.js + jsbn.js need to work in browsers without CommonJS support.
- [x] Move Python test server functionality to Node.js.
- [x] Check/update WebSockets tests.
- [x] Check/update WebSocket+WebID tests. [expired test cert, missing webid server] (did some work, not updating futher at this time)
- [x] Remove as much old build system cruft as possible.
- [x] Add some (not too slow) nodejs and web tests with forge.options.usePureJavaScript===true.
- [x] webpack vs node/browserify test count.
- [x] pkcs7 test: FIXME for date > 2050 [tests added and skipped in PhantomJS due to Date bug]
- [x] Add a CHANGELOG
- [x] Add security considerations section to README
- [x] .editorconfig
- [x] Check npm pack.
- [x] Rename 0.7.x branch to 0.8.x.

As this is a rather major change it needs testing and feedback before hitting master.  At a minimum the bower issue needs solving.  Please test and leave comments, thanks!

### CODE REVIEW

**Comment by davidlehn** [comment]
Updates:
* Lots of test refactoring.  Moved everything around and updated (mostly) everything to (mostly) work.
* Updated Node.js test server to run more tests.  Has http, https, and basic flash policy server support.  Almost free of python deps with possible exception of adding ssl session support to node test server.
* Misc things all over.

**Comment by davidlehn** [comment]
Updates:
* Lots of build system cleanup.  The old cruft is gone.
* Python test server gone and using Node.js now.
* Flash support cleaned up, builds work, and tests work.
* WebSocket support cleaned up and tests work.
* Browserify testing added.

**Comment by davidlehn** [comment]
All the tasks planned for 0.7.0 are done with one exception. The `forge-dist` repository files are written but creating that repository and doing git/bower magic is waiting for this PR to be ready to merge.  Please review and provide feedback if this PR does or does not work for your use cases.  Thanks.

**Comment by jacopotarantino** [comment]
So much approval! 👍

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+- Add "Donations" section to README.
+- Add IRC to "Contact" section of README.
+- Add "Security Considerations" section to README.
+- Add pbkdf2 usePureJavaScript test.
+- Add rsa.generateKeyPair async and usePureJavaScript tests.
+- Add .editorconfig support.
+
+### Removed
+
+- Can no longer call `forge({...})` to create new instances.
```
Comment (dlongley) [inline CHANGELOG.md]: Should include a `**BREAKING**` prefix.

**Pull request merged**